core/src/main/java/hudson/model/Node.java

```diff
@@ -214,8 +217,13 @@ public final VirtualChannel getChannel() {
 
     /**
      * Creates a new {@link Computer} object that acts as the UI peer of this {@link Node}.
+     * 
      * Nobody but {@link Jenkins#updateComputerList()} should call this method.
+     * @return Created instance of the computer.
+     *         Can be {@code null} if the {@link Node} implementation does not support it (e.g. {@link Cloud} agent).
```

Comment:
Is this actually true? A `Cloud` agent would ultimately be a `Slave`, and that is now documented to return non-null—which seems to contradict your statement that some implementations do return non-null.